crates/net/nat/src/lib.rs

```diff
@@ -185,6 +189,7 @@ pub async fn external_addr_with(resolver: NatResolver) -> Option<IpAddr> {
     match resolver {
         NatResolver::Any | NatResolver::Upnp | NatResolver::PublicIp => resolve_external_ip().await,
         NatResolver::ExternalIp(ip) => Some(ip),
+        NatResolver::NetIf => resolve_net_if_ip(DEFAULT_NET_IF_NAME).ok(),
```

Comment:
@emhane I've decided to return `None` here if resolution fails given IP resolution here is touted as best effort similarly to how both `resolve_external_ip()` above behaves in that there is no Err if that fails. 

Don't think we should be changing the signature of this method to `Result<Option<IpAddr>, NetInterfaceError>` since that would leak resolver specific impl detail to consumers. LMKWYT

Reply:
defo think sig should change. becomes hard to use this tool otherwise if error isn't printed to log. the error could be logged here too, but it's wrong to use the`reth::cli` target here, and users should be able to read the error without enabling target `reth::net`. @shekhirin will error log level be written to stdout regardless of target in default node logging setting?

Reply:
Yeah think I'd much rather log an error here or lower, than force consumers of `external_addr_with` to handle a `NetInterfaceError` even if their NatResolver impl is something unrelated WDYT 

Similar reasoning to how `resolve_external_ip()` discards the error, would be nice to have it log an error instead

Reply:
ye, would like it to panic already in cli as well if the error is logged, so if none is returned when the net if ip resolution flag is passed

please open an issue for `resolve_external_ip()` not logging error